cwatch.php

```diff
@@ -908,49 +908,53 @@ private function getMalwareTab($template, $service, $post)
 
         // Load the helpers required for this view
         Loader::loadHelpers($this, ['Form', 'Html']);
+        Loader::loadComponents($this, ['Security']);
 
         // Get cWatch API
         $api = $this->getApi();
         $service_fields = $this->serviceFieldsToObject($service->fields);
 
         if (isset($post['test_ftp'])) {
-            $use_ftps = isset($post['ftps']);
-            $error_message = Language::_('CWatch.!error.ftp' . ($use_ftps ? 's' : '') . '_test', true);
+            $use_sftp = isset($post['sftp']);
+            $error_message = Language::_('CWatch.!error.' . ($use_sftp ? 's' : '') . 'ftp_test', true);
 
             try {
-                // Set regular FTP options
-                $ftp_options = array(
-                    'passive' => true,
-                    'port' => $this->Html->ifSet($post['port']),
-                    'timeout' => 30,
-                    'curlOptions' => array(
-                        CURLOPT_PROTOCOLS => CURLPROTO_FTP,
-                    )
-                );
-                $protocol = 'ftp://';
-
-                // Update options for FTPS
-                if ($use_ftps) {
-                    $protocol = 'ftps://';
-                    $ftp_options['curlOptions'][CURLOPT_PROTOCOLS] = CURLPROTO_FTPS;
-                    $ftp_options['curlOptions'][CURLOPT_SSL_VERIFYPEER] = true;
-                    $ftp_options['curlOptions'][CURLOPT_USE_SSL] = CURLUSESSL_ALL;
-                    $ftp_options['curlOptions'][CURLOPT_FTPSSLAUTH] = CURLFTPAUTH_DEFAULT;
-                }
+                if ($use_sftp) {
+                    $ftp = $this->Security->create(
+                        'Net',
+                        'SFTP',
+                        [$post['host'], $post['port']]
+                    );
 
-                $ftp = new Ftp();
-                $ftp->setServer($protocol . $this->Html->ifSet($post['host']));
-                $ftp->setCredentials(
-                    $this->Html->ifSet($post['username']),
-                    $this->Html->ifSet($post['password'])
-                );
-                $ftp->setOptions($ftp_options);
+                    $success = $ftp->login($post['username'], $post['password']) && $ftp->chdir($post['path']);
+                } else {
+                    // Set regular FTP options
+                    $ftp_options = array(
+                        'passive' => true,
+                        'port' => $this->Html->ifSet($post['port']),
+                        'timeout' => 30,
+                        'curlOptions' => array(
+                            CURLOPT_PROTOCOLS => CURLPROTO_FTP,
+                        )
+                    );
+                    $protocol = 'ftp://';
+
+                    $ftp = new Ftp();
```

Comment:
Where is the Ftp class loaded?

Reply:
In a use statement at the top since it is namespaced